dask_cudf/io/tests/test_csv.py

```diff
@@ -27,4 +28,27 @@ def test_read_csv_w_bytes(tmp_path):
 
     result = df2.compute().to_pandas()
     expected = df.compute()
+    assert df2.npartitions is 3
     dd.assert_eq(result, expected, check_index=False)
+
+def test_read_csv_compression(tmp_path):
+    df = dask.datasets.timeseries(dtypes={"x": int, "y": int}, freq="120s").reset_index(
+        drop=True
+    )
+    df = pd.DataFrame(dict(x=np.arange(20), y=np.arange(20)))
+    ddf = dd.from_pandas(df, npartitions=1)
+    df.to_csv(tmp_path / "data.csv.gz", index=False, compression='gzip')
+
+    with pytest.warns(UserWarning) as w:
+        df2 = dask_cudf.read_csv(tmp_path / "*.csv.gz", chunksize="50 B", compression='gzip')
+
+    assert len(w) == 1
+    msg = str(w[0].message)
+    assert 'gzip' in msg
+
+    df2 = dask_cudf.read_csv(tmp_path / "*.csv.gz", chunksize=None, compression='gzip')
+    assert df2.npartitions is 1
+
+    result = df2.compute().to_pandas()
+    expected = ddf.compute()
+    dd.assert_eq(result, expected, check_index=False)
```

Comment:
My hope is that you can drop the `.compute` and `.to_pandas` calls here.  The `assert_eq` functions prefer to get unevaluated results so that they can run a bunch of sanity checks on the Dask objects before computing.  

Sometimes we call compute or to_pandas if we also need to use pandas functions to normalize the values before comparison.  This is unfortunate when it has to be done though, and is best avoided.